mesonbuild/compilers/cuda.py

```diff
@@ -34,14 +34,18 @@ def __init__(self, exelist, version, for_machine: MachineChoice, is_cross, exe_w
         super().__init__(exelist, version, for_machine, **kwargs)
         self.is_cross = is_cross
         self.exe_wrapper = exe_wrapper
-        self.id = 'nvcc'
+        self.id = CudaCompiler.cuda_id()
         default_warn_args = []
         self.warn_args = {'0': [],
                           '1': default_warn_args,
                           '2': default_warn_args + ['-Xcompiler=-Wextra'],
                           '3': default_warn_args + ['-Xcompiler=-Wextra',
                                                     '-Xcompiler=-Wpedantic']}
 
+    @staticmethod
+    def cuda_id():
+        return 'nvcc'
```

Comment:
I'd really prefer not to do this, I've got some cleanup work in progress that moves the id to the class level instead of the instance level so that you never need to set `self.id`